File: lab3/q3.cpp
// clang++ -std=c++17 -Wall -Werror -Wextra -Wpedantic -g3 -o linelength linelength.cpp
// cd r28chang/SYDE-121/assignment01

#include <iostream>
#include <string>
#include <cmath>
#include <vector>

using namespace std;

auto main() -> int
{
    std::string curr;
    double lines = 0;
    int rows = 0;
    int cols = 0;
    float currInt = 0.0;

    std::vector<float> values;

    while (getline(std::cin, curr))
    {
        lines++;
        currInt = stof(curr);

        if (lines == 1)
        {
            rows = stoi(curr);
        }
        else if (lines == 2)
        {
            cols = stoi(curr);
        }
        else
        {
            values.push_back(currInt);
        }
    }

    int counter = 0;
    vector<vector<float>> list(rows, vector<float>(cols, 0.0));

    for (int i = 0; i < rows; i++)
    {
        for (int j = 0; j < cols; j++)
        {
            list[i][j] = values[counter];
            counter++;
        }
    }

    for (int i = 0; i < rows; i++)
    {
        for (int j = 0; j < cols; j++)
        {
            std::cout << std::to_string(list[i][j]) << std::endl;
        }
    }

    return 0;
}

File: lab3/q3.py
import sys 

values = []
#list = [][] #matrix (2d array)
lines = 0 
rows = 0.0 
cols = 0.0 

for curr in sys.stdin: 
    lines += 1 
    curr_int = float(curr)

    if (lines == 1):
        rows = int(curr)
    elif (lines == 2):
        cols = int(curr)

    else: 
        values.append(curr_int)

counter = 0 
list = [[0 for x in range(cols)] for y in range(rows)] 

for i in range (0, rows, 1):
    for j in range(0, cols, 1):
        list[i][j] = values[counter]
        counter+=1  

for element in list:
    print(element)